mozilla-central/setup

```diff
@@ -109,40 +109,56 @@ if [ -n "$INDEXED_GIT_REV" ]; then
 
     date
 
-    # Download the bugzilla components file
-    ${CURL} https://index.taskcluster.net/v1/task/gecko.v2.$REVISION.source.source-bugzilla-info/artifacts/public/components-normalized.json > bugzilla-components.json
-
-    # Download the artifacts from taskcluster for each platform that we're indexing
+    # Download the bugzilla components file and the artifacts from each platform that
+    # we're indexing. But do them in parallel by emitting all the curl commands into
+    # a file and then feeding it to GNU parallel.
+    echo "${CURL} https://index.taskcluster.net/v1/task/gecko.v2.$REVISION.source.source-bugzilla-info/artifacts/public/components-normalized.json > bugzilla-components.json" > downloads.lst
     for PLATFORM in linux64 macosx64 win64 android-armv7; do
-        date
-
-        # First check that the the searchfox job exists for the platform and revision we want. Otherwise emit a warning and skip it.
+        # First check that the the searchfox job exists for the platform and revision we want. Otherwise emit a warning and skip it. This
+        # file is small so it's cheap to download and spew to stdout as a check that the analysis data for the platform exists.
         ${CURL} https://index.taskcluster.net/v1/task/gecko.v2.$REVISION.firefox.$PLATFORM-searchfox-debug/artifacts/public/build/target.json ||
         (   echo "WARNING: Unable to find analysis for $PLATFORM for hg rev $INDEXED_HG_REV; skipping analysis merge step for this platform." &&
             continue
         )
 
-        echo "Downloading artifacts for $PLATFORM..."
         TC_PREFIX="https://index.taskcluster.net/v1/task/gecko.v2.${REVISION}.firefox.${PLATFORM}-searchfox-debug/artifacts/public/build"
+        # C++ analysis
+        echo "${CURL} ${TC_PREFIX}/target.mozsearch-index.zip > ${PLATFORM}.mozsearch-index.zip" >> downloads.lst
```

Comment:
I think you could just put the urls and do xargs CURL, but there's not a huge difference so fine to leave as is.